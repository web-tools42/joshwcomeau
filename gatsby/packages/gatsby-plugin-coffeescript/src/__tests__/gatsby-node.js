jest.mock(`../resolve`, () => module => `/resolved/path/${module}`)

const {
  resolvableExtensions,
  onCreateWebpackConfig,
  preprocessSource,
} = require(`../gatsby-node`)

describe(`gatsby-plugin-coffeescript`, () => {
  it(`contains coffee script extensions`, () => {
    expect(resolvableExtensions()).toMatchSnapshot()
  })

  it(`modifies webpack config with cofeescript extensions`, () => {
    const actions = {
      setWebpackConfig: jest.fn(),
    }
    const loaders = { js: () => `babel-loader` }

    onCreateWebpackConfig({ actions, loaders })

    expect(actions.setWebpackConfig).toHaveBeenCalledTimes(
      resolvableExtensions().length
    )

    const lastCall = actions.setWebpackConfig.mock.calls.pop()
    expect(lastCall).toMatchSnapshot()
  })

  describe(`pre processing`, () => {
    it(`returns null if non-coffeescript file`, () => {
      expect(
        preprocessSource({
          filename: `test.js`,
          contents: `alert('hello');`,
        })
      ).toBe(null)
    })

    it(`transforms .coffee files`, () => {
      expect(
        preprocessSource(
          {
            filename: `test.coffee`,
            contents: `alert "I knew it!" if elvis?`,
          },
          {}
        )
      ).toMatchSnapshot()
    })
  })
})
