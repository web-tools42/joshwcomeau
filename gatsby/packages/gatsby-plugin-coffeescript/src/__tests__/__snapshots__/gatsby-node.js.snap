// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-coffeescript contains coffee script extensions 1`] = `
Array [
  ".coffee",
]
`;

exports[`gatsby-plugin-coffeescript modifies webpack config with cofeescript extensions 1`] = `
Array [
  Object {
    "module": Object {
      "rules": Array [
        Object {
          "test": /\\\\\\.coffee\\$/,
          "use": Array [
            "babel-loader",
            "/resolved/path/coffee-loader",
          ],
        },
      ],
    },
  },
]
`;

exports[`gatsby-plugin-coffeescript pre processing transforms .coffee files 1`] = `
"(function() {
  if (typeof elvis !== \\"undefined\\" && elvis !== null) {
    alert(\\"I knew it!\\");
  }

}).call(this);
"
`;
